### PR TITLE
Devoncarew locate sdk

### PR DESCRIPTION
**Pull request by devoncarew**
Use grinder's function to locate the dart sdk. It tries to find it via the --dart-sdk cli arg, by a DART_SDK env var, relative to the dart executable, and via the unix `which` command.

@keertip

### CODE REVIEW

File: lib/dartdoc.dart [hunk 1]
```diff
@@ -62,11 +65,19 @@ class DartDoc {
...
 
   List<LibraryElement> parseLibraries(List<String> files) {
-    DartSdk sdk = new DirectoryBasedDartSdk(new JavaFile(_getSdkDir().path));
+    Directory sdkDir = grinder.getSdkDir(_cliArgs);
+    if (sdkDir == null) {
+      print(
+          "Warning: unable to locate the Dart SDK. Please use the --dart-sdk "
+          "command line option or set the DART_SDK environment variable.");
+    }
+    DartSdk sdk = new DirectoryBasedDartSdk(new JavaFile(sdkDir.path));
```
Comment (sethladd) [inline lib/dartdoc.dart]: what is `new JavaFile` ?

File: bin/dartdoc.dart [hunk 2]
```diff
@@ -23,7 +23,8 @@ void main(List<String> arguments) {
 
   String url = results['url'];
   var currentDir = Directory.current;
-  new DartDoc(currentDir, excludeLibraries, url).generateDocs();
+  DartDoc dartdoc = new DartDoc(currentDir, excludeLibraries, url, arguments);
+  dartdoc.generateDocs();
```
Comment (sethladd) [inline bin/dartdoc.dart]: method cascades?

File: lib/dartdoc.dart (hunk 1 above)
Comment (devoncarew) [inline lib/dartdoc.dart]: It's a weird `analytics` engine-ism.

File: bin/dartdoc.dart (hunk 2 above)
Comment (devoncarew) [inline bin/dartdoc.dart]: ok

File: lib/dartdoc.dart (hunk 1 above)
Comment (sethladd) [inline lib/dartdoc.dart]: yuck. we should change that in the analysis engine.

**Comment by sethladd** [comment]
lgtm

File: lib/dartdoc.dart (hunk 1 above)
Comment (devoncarew) [inline lib/dartdoc.dart]: Set your geologic clock.

**Comment by keertip** [comment]
lgtm! Thanks!

**Comment by devoncarew** [comment]
Hmm, looks like I was working off a slightly older version of the source :( Have merged to master.

**Pull request merged**